include/layers/FCLayer.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+#include <stdexcept>
+#include <vector>
+
+typedef double ValueType;
+
+using namespace std;
+
+vector<ValueType> mat_vec_mul(const vector<vector<ValueType> >& mat,
+                              const vector<ValueType>& vec);
+
+class FCLayer {
```

Comment:
It makes sense to establish abstract layer class and inherit FC layer from it

Reply:
Yrd